File: examples/arrayperformance.cs
using System;
using Boo.Lang;

public class App
{
	public static void Main()
	{
		Test();
		Test();
		Test();
	}
	
	private static void Test()
	{
		const int items = 2000000;

		object[] array = (object[])new List(Builtins.range(items)).ToArray(typeof(object));
		
		List collect = new List();

		DateTime start = DateTime.Now;
		foreach (int i in Builtins.range(items))
		{
			collect.Add(array[i]);
		}
		TimeSpan elapsed = DateTime.Now.Subtract(start);

		Console.WriteLine("{0} elapsed.", elapsed.TotalMilliseconds);
	}
}

File: examples/misc/arrayperformance.py
from time import time

def test():
	items = 2000000
	
	array = tuple(range(items))
	
	collect = []
	
	start = time() 
	for i in xrange(items):
		collect.append(array[i])	
	elapsed = time() - start
	
	print elapsed*1000, " elapsed."
	
test()
test()
test()
